Basket-Donation-Payments/Basket-Donation-Payments/View Controllers/Profile View/NonprofitContainerViewController.m

```diff
@@ -27,17 +27,25 @@ @implementation NonprofitContainerViewController
 
 - (void)viewDidLoad {
     [super viewDidLoad];
-    self.foundInBasketsTableView.delegate = self;
-    self.foundInBasketsTableView.dataSource = self;
-    self.nonprofit = [[User currentUser] nonprofit];
-//    if (self.nonprofit.verificationFiles == nil) { //FIXME: ISVERIFIED
-//        self.nonprofitIsVerifiedCheckmark.hidden = YES;
-//    }
-    self.nonprofitNameLabel.text = self.nonprofit.nonprofitName;
-    self.nonprofitCategoryLabel.text = self.nonprofit.category;
-    self.totalDonationValueLabel.text = [NSString stringWithFormat:@"$%0.2f", self.nonprofit.totalDonationsValue];
-    self.nonprofitWebsiteLabel.text = self.nonprofit.websiteUrlString;
-    self.nonprofitDescriptionLabel.text = self.nonprofit.nonprofitDescription;
+    if ([[User currentUser] nonprofit] != nil) {
+//        self.foundInBasketsTableView.delegate = self;
+//        self.foundInBasketsTableView.dataSource = self;
+        self.nonprofit = [[User currentUser] nonprofit];
+        
+        NSPredicate *pred = [NSPredicate predicateWithFormat:@"objectId == %@", self.nonprofit.objectId];
+        PFQuery *thisNonprofitQuery = [PFQuery queryWithClassName:@"Nonprofit" predicate:pred];
+        self.nonprofit = [thisNonprofitQuery findObjects][0];
```

Comment:
You should always check for out of boundary exceptions before accessing array elements, or use `firstObject` instead of `[0]` which will return `nil` if array is empty instead of throwing an exception and crashing the app.